orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTaskSpec.groovy

```diff
@@ -146,15 +146,65 @@ class MonitorKatoTaskSpec extends Specification {
     "about to fail"      | MonitorKatoTask.TASK_NOT_FOUND_TIMEOUT     |  true         | 'N/A'
   }
 
+  def "should set kato.task.skipReplica=true when getting a task from clouddriver times out"() {
+    given:
+    def taskId = "katoTaskId"
+    def elapsed = MonitorKatoTask.TASK_NOT_FOUND_TIMEOUT + 1
+    task.kato = Mock(KatoService)
+
+    def stage = new Stage(Execution.newPipeline("orca"), "whatever", [
+      "kato.last.task.id": new TaskId(taskId),
+      "kato.task.firstNotFoundRetry": now.minusMillis(elapsed).toEpochMilli()
+    ])
+
+    assert !(stage.context['kato.task.skipReplica'] as Boolean)
+
+    when:
+    def result = task.execute(stage)
+
+    then:
+    result.context['kato.task.skipReplica'] as Boolean
+    notThrown(RetrofitError)
+    with(task.kato) {
+      1 * lookupTask(taskId, false) >> { retrofit404() }
+      0 * lookupTask(taskId, true)
+    }
+  }
+
+  def "should get task from master when kato.task.skipReplica=true"() {
+    given:
+    def taskId = "katoTaskId"
+    def elapsed = MonitorKatoTask.TASK_NOT_FOUND_TIMEOUT + 1
+    task.kato = Mock(KatoService)
+    def stage = new Stage(Execution.newPipeline("orca"), "whatever", [
+      "kato.last.task.id": new TaskId(taskId),
+      "kato.task.firstNotFoundRetry": now.minusMillis(elapsed).toEpochMilli(),
+      "kato.task.skipReplica": true
+    ])
+
+    when:
+    task.execute(stage)
+
+    then:
+    notThrown(RetrofitError)
+    with(task.kato) {
+      1 * lookupTask(taskId, true) >> Observable.from(new Task(taskId, new Task.Status(completed: true, failed: false), [], []))
+      0 * lookupTask(taskId, false)
+    }
+  }
+
   def "should timeout if task not not found after timeout period"() {
     given:
+    def taskId = "katoTaskId"
     def ctx = [
       "kato.last.task.id": new TaskId(taskId)
     ]
     ctx.put('kato.task.firstNotFoundRetry', now.minusMillis(elapsed).toEpochMilli())
+    ctx.put('kato.task.skipReplica', true)
     def stage = new Stage(Execution.newPipeline("orca"), "whatever", ctx)
     task.kato = Stub(KatoService) {
-      lookupTask(taskId) >> { retrofit404() }
+      lookupTask(taskId, false) >> { retrofit404() }
```

Comment:
If we're setting `skipReplica == true` should we ever even expect a call to `lookupTask(taskId, false)`